spring-amqp/src/main/java/org/springframework/amqp/core/DeclarableCustomizer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.core;
+
+import java.util.function.Function;
+
+/**
+ * Beans of this type are invoked by the {@link AmqpAdmin} before declaring the
+ * {@link Declarable}, allowing customization thereof.
+ *
+ * @author Gary Russell
+ * @since 2.2.2
+ *
+ */
+@FunctionalInterface
+public interface DeclarableCustomizer extends Function<Declarable, Declarable> {
```

Comment:
Why `function`, but not `consumer`?
Are you intended to replace property with result of this customizer?

Reply:
Yes; they can replace the object if they need to for some reason (aside from the arguments, they are mostly immutable).